qiskit_experiments/framework/base_experiment.py

```diff
@@ -146,6 +134,37 @@ def run(
         # Return the ExperimentData future
         return experiment_data
 
+    def _initialize_experiment_data(
+        self, backend: Backend, experiment_data: Optional[ExperimentData] = None
+    ) -> ExperimentData:
+        """Initialize the return data container for the experiment run"""
+        if experiment_data is None:
+            return self.__experiment_data__(experiment=self, backend=backend)
+
+        # Validate experiment is compatible with existing data
+        metadata = experiment_data.metadata()
+        if metadata.get("experiment_type") != self._type:
+            raise QiskitError("Existing ExperimentData contains data from a different experiment.")
+        if metadata.get("physical_qubits") != list(self.physical_qubits):
+            raise QiskitError(
+                "Existing ExperimentData contains data for a different set of physical qubits."
+            )
+
+        # Generate new ExperimentData containing a copy of previous
+        # jobs and data, but not analysis results or figures
+        ret = self.__experiment_data__(experiment=self, backend=backend)
```

Comment:
What if you copy the experiment data object and just override id?

Reply:
Theres a lot of internal attributes in the DbExperimentData class that I don't completely understand, so I wasn't sure how safe this would be. @jyu00 prob knows best, could we add a method to that class that makes a duplicate with a new ID and no analysis results?